MidCamp/midcamp
# Featured speakers

# Description

This satisfies the requirement:

> Make sure we can feature speakers f’reals like the styleguide

From the [43 Days to Midcamp document](https://docs.google.com/document/d/1NfmSQnOiYbkHHe8j6jZFboIjwQbFTaJZO48Ui_27q2I/edit#).  It includes:

- Featured Speaker view mode for User entities
- Theme hook suggestions for the new view mode
- The entityqueue module
- An entityqueue for adding Featured Speakers
- A view for controlling the entityqueue and showing Featured Speakers on the homepage
- Preprocessing for theming variables
- Template files for the view mode and its field

# To test

- `composer install`
- Import config with `drush cim -y`
- Clear caches with `drush cr`
- Visit the entityqueue at http://midcamp.org.docker.amazee.io/admin/structure/entityqueue/featured_speakers/featured_speakers.  Add some Users with the "Speaker" role (up to 3)
- Visit the homepage.  Observe that the speakers display.

![image](https://user-images.githubusercontent.com/4048700/35711400-4b9f4fc0-0782-11e8-838d-a2e39fc34bd1.png)

# Notes

I submitted a PR in the theme (https://github.com/MidCamp/Hatter/pull/39) to remove the background swirl, because it relied on a transparent, half black and white, half colorized sprite to work correctly.
![featured-speaker-styleguide](https://user-images.githubusercontent.com/4048700/35711455-9686bd0c-0782-11e8-8da4-db18bc6ca5d5.png)

That looks really cool but I didn't have a lot of time to think of nice way to fix it so things like this didn't happen:
![featured-speaker-drupal-image](https://user-images.githubusercontent.com/4048700/35711477-b24cc202-0782-11e8-89a6-90e5c6ab638a.png)
Or even this when no image was present:
![featured-speaker-drupal-no-image](https://user-images.githubusercontent.com/4048700/35711485-b943c27c-0782-11e8-9f47-8a38f6bd9ee9.png)
For now I've settled for no background image at all.
